feature-libs/customer-ticketing/components/customer-ticketing-details.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import {
+  CustomerTicketingFacade,
+  STATUS,
+  Status,
+  TicketDetails,
+  TicketEvent,
+} from '@spartacus/customer-ticketing/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+} from '@spartacus/core';
+import { LaunchDialogService } from '@spartacus/storefront';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CustomerTicketingDetailsService {
+  dataLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  protected ticketDetails$ = this.customerTicketingService.getTicket();
+
+  constructor(
+    protected customerTicketingService: CustomerTicketingFacade,
+    protected globalMessageService: GlobalMessageService,
+    protected launchDialogService: LaunchDialogService,
+    protected routingService: RoutingService
+  ) {}
+
+  getTicketSubject = (): Observable<string | undefined> =>
+    this.ticketDetails$.pipe(map((details) => details?.subject));
+
+  getTicketStatus = (): Observable<string | undefined> =>
+    this.ticketDetails$.pipe(
+      map((details) => details?.status?.id?.toUpperCase())
+    );
+
+  getAvailableTransitionStatus = (): Observable<Status[] | undefined> =>
+    this.ticketDetails$.pipe(
+      map((details) => details?.availableStatusTransitions)
+    );
+
+  getTicketDetails = (): Observable<TicketDetails | undefined> =>
+    this.ticketDetails$;
+
+  getTicketState() {
+    this.customerTicketingService.getTicketState().subscribe();
+  }
+
+  createTicketEvent(ticketEvent: TicketEvent): void {
+    this.dataLoading$.next(true);
+    this.customerTicketingService.createTicketEvent(ticketEvent).subscribe({
+      complete: () => {
+        this.customerTicketingService.getTicketState().subscribe((state) => {
```

Comment:
this nested subscription is causing a side effect to when you close the ticket. 

It is calling the global message that says it is closed and reopen.

To be honest, I'm a little confused here. Let's discuss the train of thought and improve as we go in a call

Reply:
The logic I would have moved it to the component level than in the service to handle the state of what is supposed to happen